app/services/qa/linked_data/mapper/graph_mapper_service.rb

```diff
@@ -29,9 +29,9 @@ def self.map_values(graph:, predicate_map:, subject_uri:)
           value_map = {}
           predicate_map.each do |key, predicate|
             values = predicate == :subject_uri ? [subject_uri] : graph_service.object_values(graph: graph, subject: subject_uri, predicate: predicate)
-            values = yield values if block_given?
             value_map[key] = values
           end
+          value_map = yield value_map if block_given?
```

Comment:
This should have been here to allow appending extra keys to the set instead of trying to effect a single value in the set.  This will be used to append the context to the set.